src/base/bittorrent/session.cpp

```diff
@@ -1656,8 +1664,9 @@ void Session::saveResumeData()
                     torrent->handleAlert(a);
                 break;
             }
```

Comment:
There is a foreach "hidden" in the diff. Shouldn't you change it to for() like you did in readAlerts()?